tests/Feature/GroupBalancedDivisionTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+use App\Helpers\GroupBalancedDivision;
+use App\Models\Event;
+use App\Models\User;
+use App\Models\Course;
+use App\Models\Registration;
+use App\Models\Group;
+
+
+uses(\Illuminate\Foundation\Testing\RefreshDatabase::class);
+
+test('equal div only drinkers', function () {
+    $event = Event::factory()->create(['consider_alcohol' => true])->first();
+    $courses = Course::factory()->count(10)->create();
+    $groups = Group::factory()->count(10)->create(['event_id' => $event->id]);
+
+    foreach ($courses as $course) {
+        User::factory()->count(20)->create(['course_id' => $course->id]);
```

Comment:
I doubt whether this is a valid test method, in which you create the same number of users for each couirse in chronological order. It would be better to create e.g. 100 users with random courses and then see whether this is balanced. This can be done by chance so that they are simply assigned one after the other and are thus balanced.